src/framework/vst/internal/vstplugin.cpp

```diff
@@ -0,0 +1,123 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "vstplugin.h"
+
+#include "log.h"
+
+using namespace mu;
+using namespace mu::vst;
+
+VstPlugin::VstPlugin()
+    : m_factory(nullptr)
+{
+}
+
+Ret VstPlugin::load(const std::string& pluginPath)
+{
+    PluginContextFactory::instance().setPluginContext(&m_pluginContext);
+
+    std::string errorString;
+
+    m_module = PluginModule::create(pluginPath, errorString);
+
+    if (!m_module) {
+        LOGE() << errorString;
+        return make_ret(Err::NoPluginModule);
+    }
+
+    m_factory = m_module->getFactory();
+
+    if (!m_factory.get()) {
+        return make_ret(Err::NoPluginFactory);
+    }
+
+    for (const ClassInfo& classInfo : m_factory.classInfos()) {
+        if (classInfo.category() != kVstAudioEffectClass) {
+            continue;

```

Comment:
I think would be like add LOGI message, about that audio not supported now

Reply:
agreed